src/actions/ast.js

```diff
@@ -126,3 +138,64 @@ export function setSelection(
     });
   };
 }
+
+/**
+ * Action dispatched from stepOver action. Receives the position
+ * and depending on where it is paused, sets up hidden breakpoints
+ * and handles resume, stepOver (as needed)
+ * @param position
+ * @returns {function(ThunkArgs)}
+ */
+export function analyzeStepping(type) {
+  return async ({ dispatch, getState, client }: ThunkArgs) => {
+    const position = await dispatch(getPosition());
+    const source = getSelectedSource(getState()).toJS();
+    const hiddenBreakpointLocation = getHiddenBreakpoint(getState());
+
+    await dispatch({
+      type: "ANALYZE_STEPPING",
+      [PROMISE]: (async function() {
+        if (hiddenBreakpointLocation) {
+          await dispatch(removeBreakpoint(hiddenBreakpointLocation));
+        }
+        const path = getClosestPath(source, position);
```

Comment:
you're currently using `getClosestPath` directly by requiring it. Unfortunately, we need to do the search on the parser worker, so we'll have to move some of these methods to the worker.

here are some [docs](https://github.com/devtools-html/debugger.html/blob/master/docs/local-development.md#adding-a-task) on adding a new worker task.

I think we'll need to add tasks like `isAwaitExpression` or `insideAsyncFunction` to the worker. We could avoid this if those helpers could use a babel node. The distinction is that Nodes don't have prototypes. Paths do.